test/functional/http/post/1.second.secret.js

```diff
@@ -0,0 +1,388 @@
+
+	var badTransactions = [];
+	var goodTransactions = [];
+	var badTransactionsEnforcement = [];
+	var goodTransactionsEnforcement = [];
+	var pendingMultisignatures = [];
+
+	var account = node.randomAccount();
+	var accountNoFunds = node.randomAccount();
+	var accountMinimalFunds = node.randomAccount();
+	var accountNoSecondPassword = node.randomAccount();
+	var accountDuplicate = node.randomAccount();
+
+	var transaction, signature;
+
+	// Crediting accounts
+	before(function () {
+		var promises = [];
+		promises.push(creditAccountPromise(account.address, 100000000000));
+		promises.push(creditAccountPromise(accountMinimalFunds.address, constants.fees.secondsignature));
+		promises.push(creditAccountPromise(accountNoSecondPassword.address, constants.fees.secondsignature));
+		promises.push(creditAccountPromise(accountDuplicate.address, constants.fees.secondsignature));
+
+		return node.Promise.all(promises).then(function (results) {
+			results.forEach(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').that.is.not.empty;
+			});
+		}).then(function (res) {
+			return onNewBlockPromise();
+		});
+	});
+
+	describe('schema validations', function () {
+
+		shared.invalidAssets(account, 'signature', badTransactions);
+	});
+
+	describe('transactions processing', function () {
+
+		it('with no funds should fail', function () {
+			transaction = node.lisk.signature.createSignature(accountNoFunds.password, accountNoFunds.secondPassword);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.not.be.ok;
+				node.expect(res).to.have.property('message').to.equal('Account does not have enough LSK: ' + accountNoFunds.address + ' balance: 0');
+				badTransactions.push(transaction);
+			});
+		});
+
+		it('with minimal required amount of funds should be ok', function () {
+			transaction = node.lisk.signature.createSignature(accountMinimalFunds.password, accountMinimalFunds.secondPassword);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+
+		it('with valid params should be ok', function () {
+			transaction = node.lisk.signature.createSignature(account.password, account.secondPassword, 1);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+	});
+
+	describe('unconfirmed state', function () {
+
+		describe('type 0 - sending funds', function () {
+
+			it('using a second passphrase on an account wih second passphrase in process should fail', function () {
```

Comment:
using second signature with an account that has a pending second passphrase registration should fail